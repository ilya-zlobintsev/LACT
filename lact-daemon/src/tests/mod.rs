use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
