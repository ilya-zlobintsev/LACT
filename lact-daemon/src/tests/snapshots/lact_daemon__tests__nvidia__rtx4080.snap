---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": null,
  "info": {
    "driver": "nvidia",
    "link_info": {
      "current_speed": "5.0 GT/s PCIe",
      "current_width": "16"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "AD103 [GeForce RTX 4080]",
        "model_id": "2704",
        "vendor": "NVIDIA Corporation",
        "vendor_id": "10DE"
      },
      "subsystem_pci_info": {
        "model_id": "5110",
        "vendor": "Micro-Star International Co., Ltd. [MSI]",
        "vendor_id": "1462"
      }
    }
  },
  "pci_info": {
    "device_pci_info": {
      "model": "AD103 [GeForce RTX 4080]",
      "model_id": "2704",
      "vendor": "NVIDIA Corporation",
      "vendor_id": "10DE"
    },
    "subsystem_pci_info": {
      "model_id": "5110",
      "vendor": "Micro-Star International Co., Ltd. [MSI]",
      "vendor_id": "1462"
    }
  },
  "power_profile_modes": null,
  "stats": {
    "clockspeed": {},
    "fan": {
      "control_enabled": false,
      "pmfw_info": {}
    },
    "power": {},
    "temps": {},
    "voltage": {},
    "vram": {}
  }
}
